docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,188 @@
+[[persistent-queues-limitations]]
+==== Limitations of persistent queues
+
+Persistent queues do not solve these problems:
+
+* Input plugins that do not use a request-response protocol cannot be protected from data loss. Tcp, udp, zeromq push+pull, and many other inputs do not have a mechanism to acknowledge receipt to the sender. 
+(Plugins such as beats and http, which *do* have an acknowledgement capability, are well protected by this queue.)
+* A persistent queue does not handle permanent machine failures such as disk corruption, disk failure, and machine loss. 
+The data persisted to disk is not replicated.
+
+
+[[configuring-persistent-queues]]
+==== Configuring persistent queues
+
+To configure persistent queues, specify options in the Logstash <<logstash-settings-file,settings file>>:
+
+`queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
+`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`.
+`queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
+`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
+`queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits.
+`queue.drain`:: Specify `true` if you want Logstash to wait until the persistent queue is drained before shutting down. The amount of time it takes to drain the queue depends on the number of events that have accumulated in the queue. Therefore, you should avoid using this setting unless the queue, even when full, is relatively small and can be drained quickly. 
+// Technically, I know, this isn't "maximum number of events" it's really maximum number of events not yet read by the pipeline worker. We only use this for testing and users generally shouldn't be setting this.
+`queue.max_events`:: The maximum number of events that are allowed in the queue. The default is 0 (unlimited).
+`queue.max_bytes`:: The total capacity of the queue in number of bytes. The
+default is 1024mb (1gb). Make sure the capacity of your disk drive is greater
+than the value you specify here.
++
+TIP: If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
+A smaller value produces smaller queues and improves queue performance. 
+
+`queue.checkpoint.acks`:: Sets the number of acked events before forcing a checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
+`queue.checkpoint.writes`:: Sets the maximum number of written events before a forced checkpoint. 
```

Comment:
There is a link further in the option description (line 66). I put in a brief description of trade-offs and then the link in case more info is needed. We can discuss structure and relocating.